// @neshan-maps-platform/react-openlayers/dist/React-Openlayers.js
import Le, { createContext as pr, useReducer as vr, forwardRef as Ne, useContext as mr, useRef as yr, useState as ce, useImperativeHandle as hr, useEffect as gr } from "react";
var le = {}, _r = {
  get exports() {
    return le;
  },
  set exports(s) {
    le = s;
  }
}, N = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Fe;
function Er() {
  if (Fe)
    return N;
  Fe = 1;
  var s = Le, f = Symbol.for("react.element"), P = Symbol.for("react.fragment"), b = Object.prototype.hasOwnProperty, v = s.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, F = { key: !0, ref: !0, __self: !0, __source: !0 };
  function A(O, m, x) {
    var _, T = {}, j = null, C = null;
    x !== void 0 && (j = "" + x), m.key !== void 0 && (j = "" + m.key), m.ref !== void 0 && (C = m.ref);
    for (_ in m)
      b.call(m, _) && !F.hasOwnProperty(_) && (T[_] = m[_]);
    if (O && O.defaultProps)
      for (_ in m = O.defaultProps, m)
        T[_] === void 0 && (T[_] = m[_]);
    return { $$typeof: f, type: O, key: j, ref: C, props: T, _owner: v.current };
  }
  return N.Fragment = P, N.jsx = A, N.jsxs = A, N;
}
var W = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ie;
function br() {
  return Ie || (Ie = 1, process.env.NODE_ENV !== "production" && function() {
    var s = Le, f = Symbol.for("react.element"), P = Symbol.for("react.portal"), b = Symbol.for("react.fragment"), v = Symbol.for("react.strict_mode"), F = Symbol.for("react.profiler"), A = Symbol.for("react.provider"), O = Symbol.for("react.context"), m = Symbol.for("react.forward_ref"), x = Symbol.for("react.suspense"), _ = Symbol.for("react.suspense_list"), T = Symbol.for("react.memo"), j = Symbol.for("react.lazy"), C = Symbol.for("react.offscreen"), S = Symbol.iterator, Y = "@@iterator";
    function H(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = S && e[S] || e[Y];
      return typeof r == "function" ? r : null;
    }
    var w = s.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function y(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        fe("error", e, t);
      }
    }
    function fe(e, r, t) {
      {
        var n = w.ReactDebugCurrentFrame, i = n.getStackAddendum();
        i !== "" && (r += "%s", t = t.concat([i]));
        var c = t.map(function(o) {
          return String(o);
        });
        c.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, c);
      }
    }
    var X = !1, $ = !1, Z = !1, Q = !1, ee = !1, V;
    V = Symbol.for("react.module.reference");
    function B(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === b || e === F || ee || e === v || e === x || e === _ || Q || e === C || X || $ || Z || typeof e == "object" && e !== null && (e.$$typeof === j || e.$$typeof === T || e.$$typeof === A || e.$$typeof === O || e.$$typeof === m || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === V || e.getModuleId !== void 0));
    }
    function u(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var i = r.displayName || r.name || "";
      return i !== "" ? t + "(" + i + ")" : t;
    }
    function g(e) {
      return e.displayName || "Context";
    }
    function l(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && y("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case b:
          return "Fragment";
        case P:
          return "Portal";
        case F:
          return "Profiler";
        case v:
          return "StrictMode";
        case x:
          return "Suspense";
        case _:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case O:
            var r = e;
            return g(r) + ".Consumer";
          case A:
            var t = e;
            return g(t._context) + ".Provider";
          case m:
            return u(e, e.render, "ForwardRef");
          case T:
            var n = e.displayName || null;
            return n !== null ? n : l(e.type) || "Memo";
          case j: {
            var i = e, c = i._payload, o = i._init;
            try {
              return l(o(c));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var h = Object.assign, D = 0, de, pe, ve, me, ye, he, ge;
    function _e() {
    }
    _e.__reactDisabledLog = !0;
    function Ve() {
      {
        if (D === 0) {
          de = console.log, pe = console.info, ve = console.warn, me = console.error, ye = console.group, he = console.groupCollapsed, ge = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: _e,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        D++;
      }
    }
    function Be() {
      {
        if (D--, D === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: h({}, e, {
              value: de
            }),
            info: h({}, e, {
              value: pe
            }),
            warn: h({}, e, {
              value: ve
            }),
            error: h({}, e, {
              value: me
            }),
            group: h({}, e, {
              value: ye
            }),
            groupCollapsed: h({}, e, {
              value: he
            }),
            groupEnd: h({}, e, {
              value: ge
            })
          });
        }
        D < 0 && y("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var re = w.ReactCurrentDispatcher, te;
    function U(e, r, t) {
      {
        if (te === void 0)
          try {
            throw Error();
          } catch (i) {
            var n = i.stack.trim().match(/\n( *(at )?)/);
            te = n && n[1] || "";
          }
        return `
` + te + e;
      }
    }
    var ne = !1, q;
    {
      var Ue = typeof WeakMap == "function" ? WeakMap : Map;
      q = new Ue();
    }
    function Ee(e, r) {
      if (!e || ne)
        return "";
      {
        var t = q.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      ne = !0;
      var i = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var c;
      c = re.current, re.current = null, Ve();
      try {
        if (r) {
          var o = function() {
            throw Error();
          };
          if (Object.defineProperty(o.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(o, []);
            } catch (k) {
              n = k;
            }
            Reflect.construct(e, [], o);
          } else {
            try {
              o.call();
            } catch (k) {
              n = k;
            }
            e.call(o.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (k) {
            n = k;
          }
          e();
        }
      } catch (k) {
        if (k && n && typeof k.stack == "string") {
          for (var a = k.stack.split(`
`), E = n.stack.split(`
`), d = a.length - 1, p = E.length - 1; d >= 1 && p >= 0 && a[d] !== E[p]; )
            p--;
          for (; d >= 1 && p >= 0; d--, p--)
            if (a[d] !== E[p]) {
              if (d !== 1 || p !== 1)
                do
                  if (d--, p--, p < 0 || a[d] !== E[p]) {
                    var R = `
` + a[d].replace(" at new ", " at ");
                    return e.displayName && R.includes("<anonymous>") && (R = R.replace("<anonymous>", e.displayName)), typeof e == "function" && q.set(e, R), R;
                  }
                while (d >= 1 && p >= 0);
              break;
            }
        }
      } finally {
        ne = !1, re.current = c, Be(), Error.prepareStackTrace = i;
      }
      var M = e ? e.displayName || e.name : "", De = M ? U(M) : "";
      return typeof e == "function" && q.set(e, De), De;
    }
    function qe(e, r, t) {
      return Ee(e, !1);
    }
    function ze(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function z(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return Ee(e, ze(e));
      if (typeof e == "string")
        return U(e);
      switch (e) {
        case x:
          return U("Suspense");
        case _:
          return U("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case m:
            return qe(e.render);
          case T:
            return z(e.type, r, t);
          case j: {
            var n = e, i = n._payload, c = n._init;
            try {
              return z(c(i), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var J = Object.prototype.hasOwnProperty, be = {}, Re = w.ReactDebugCurrentFrame;
    function G(e) {
      if (e) {
        var r = e._owner, t = z(e.type, e._source, r ? r.type : null);
        Re.setExtraStackFrame(t);
      } else
        Re.setExtraStackFrame(null);
    }
    function Je(e, r, t, n, i) {
      {
        var c = Function.call.bind(J);
        for (var o in e)
          if (c(e, o)) {
            var a = void 0;
            try {
              if (typeof e[o] != "function") {
                var E = Error((n || "React class") + ": " + t + " type `" + o + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[o] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw E.name = "Invariant Violation", E;
              }
              a = e[o](r, o, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (d) {
              a = d;
            }
            a && !(a instanceof Error) && (G(i), y("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, o, typeof a), G(null)), a instanceof Error && !(a.message in be) && (be[a.message] = !0, G(i), y("Failed %s type: %s", t, a.message), G(null));
          }
      }
    }
    var Ge = Array.isArray;
    function ae(e) {
      return Ge(e);
    }
    function Ke(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function He(e) {
      try {
        return Te(e), !1;
      } catch {
        return !0;
      }
    }
    function Te(e) {
      return "" + e;
    }
    function Ce(e) {
      if (He(e))
        return y("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ke(e)), Te(e);
    }
    var L = w.ReactCurrentOwner, Xe = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Pe, Se, oe;
    oe = {};
    function Ze(e) {
      if (J.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Qe(e) {
      if (J.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function er(e, r) {
      if (typeof e.ref == "string" && L.current && r && L.current.stateNode !== r) {
        var t = l(L.current.type);
        oe[t] || (y('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', l(L.current.type), e.ref), oe[t] = !0);
      }
    }
    function rr(e, r) {
      {
        var t = function() {
          Pe || (Pe = !0, y("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function tr(e, r) {
      {
        var t = function() {
          Se || (Se = !0, y("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var nr = function(e, r, t, n, i, c, o) {
      var a = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: f,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: o,
        // Record the component responsible for creating this element.
        _owner: c
      };
      return a._store = {}, Object.defineProperty(a._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(a, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(a, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: i
      }), Object.freeze && (Object.freeze(a.props), Object.freeze(a)), a;
    };
    function ar(e, r, t, n, i) {
      {
        var c, o = {}, a = null, E = null;
        t !== void 0 && (Ce(t), a = "" + t), Qe(r) && (Ce(r.key), a = "" + r.key), Ze(r) && (E = r.ref, er(r, i));
        for (c in r)
          J.call(r, c) && !Xe.hasOwnProperty(c) && (o[c] = r[c]);
        if (e && e.defaultProps) {
          var d = e.defaultProps;
          for (c in d)
            o[c] === void 0 && (o[c] = d[c]);
        }
        if (a || E) {
          var p = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          a && rr(o, p), E && tr(o, p);
        }
        return nr(e, a, E, i, n, L.current, o);
      }
    }
    var ie = w.ReactCurrentOwner, we = w.ReactDebugCurrentFrame;
    function I(e) {
      if (e) {
        var r = e._owner, t = z(e.type, e._source, r ? r.type : null);
        we.setExtraStackFrame(t);
      } else
        we.setExtraStackFrame(null);
    }
    var se;
    se = !1;
    function ue(e) {
      return typeof e == "object" && e !== null && e.$$typeof === f;
    }
    function Oe() {
      {
        if (ie.current) {
          var e = l(ie.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function or(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var xe = {};
    function ir(e) {
      {
        var r = Oe();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function je(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = ir(r);
        if (xe[t])
          return;
        xe[t] = !0;
        var n = "";
        e && e._owner && e._owner !== ie.current && (n = " It was passed a child from " + l(e._owner.type) + "."), I(e), y('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), I(null);
      }
    }
    function ke(e, r) {
      {
        if (typeof e != "object")
          return;
        if (ae(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            ue(n) && je(n, r);
          }
        else if (ue(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var i = H(e);
          if (typeof i == "function" && i !== e.entries)
            for (var c = i.call(e), o; !(o = c.next()).done; )
              ue(o.value) && je(o.value, r);
        }
      }
    }
    function sr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === m || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === T))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = l(r);
          Je(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !se) {
          se = !0;
          var i = l(r);
          y("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", i || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && y("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function ur(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            I(e), y("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), I(null);
            break;
          }
        }
        e.ref !== null && (I(e), y("Invalid attribute `ref` supplied to `React.Fragment`."), I(null));
      }
    }
    function Ae(e, r, t, n, i, c) {
      {
        var o = B(e);
        if (!o) {
          var a = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (a += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var E = or(i);
          E ? a += E : a += Oe();
          var d;
          e === null ? d = "null" : ae(e) ? d = "array" : e !== void 0 && e.$$typeof === f ? (d = "<" + (l(e.type) || "Unknown") + " />", a = " Did you accidentally export a JSX literal instead of a component?") : d = typeof e, y("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", d, a);
        }
        var p = ar(e, r, t, i, c);
        if (p == null)
          return p;
        if (o) {
          var R = r.children;
          if (R !== void 0)
            if (n)
              if (ae(R)) {
                for (var M = 0; M < R.length; M++)
                  ke(R[M], e);
                Object.freeze && Object.freeze(R);
              } else
                y("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              ke(R, e);
        }
        return e === b ? ur(p) : sr(p), p;
      }
    }
    function cr(e, r, t) {
      return Ae(e, r, t, !0);
    }
    function lr(e, r, t) {
      return Ae(e, r, t, !1);
    }
    var fr = lr, dr = cr;
    W.Fragment = b, W.jsx = fr, W.jsxs = dr;
  }()), W;
}
(function(s) {
  process.env.NODE_ENV === "production" ? s.exports = Er() : s.exports = br();
})(_r);
const K = le.jsx;
const Rr = "_map_1yxt5_2", Tr = "_touch_1yxt5_26", Cr = "_floaten_1yxt5_45", Pr = "_float_1yxt5_45", Sr = {
  map: Rr,
  "map-popup-container": "_map-popup-container_1yxt5_7",
  touch: Tr,
  "mobile-layers-button": "_mobile-layers-button_1yxt5_30",
  floaten: Cr,
  float: Pr
}, We = (s) => s ? s instanceof Object ? [s.longitude, s.latitude] : s : void 0, wr = async () => {
  try {
    const f = await new Promise(
      (P, b) => {
        navigator.geolocation.getCurrentPosition(
          (v) => {
            P(v);
          },
          (v) => b(v)
        );
      }
    );
    return f && We(f.coords) || Object.values(Me());
  } catch (s) {
    console.error(s);
    const f = Me();
    return [f.longitude, f.latitude];
  }
}, Me = () => ({
  longitude: 51.36281969540723,
  latitude: 35.69672648316882
}), Or = {
  reverse: "https://api.neshan.org/v4/reverse",
  search: "https://api.neshan.org/v1/search",
  map: "https://static.neshan.org/sdk/openlayers/5.3.0/ol.js"
}, Ye = pr({
  map: null,
  mapType: "neshan",
  mapContainer: null,
  setMap: (s) => {
  },
  setMapType: (s) => {
  },
  setMapContainer: (s) => {
  }
}), $e = {
  map: null,
  mapType: "neshan",
  mapContainer: null
}, xr = (s, f) => {
  switch (f.type) {
    case "SET_MAP":
      return {
        ...s,
        map: f.payload
      };
    case "SET_MAP_TYPE":
      return {
        ...s,
        title: f.payload
      };
    default:
      return $e;
  }
}, jr = (s) => {
  const [f, P] = vr(xr, $e), b = {
    ...f,
    setMap: (v) => {
      P({
        type: "SET_MAP",
        payload: v
      });
    },
    setMapType: (v) => {
      P({
        type: "SET_MAP_TYPE",
        payload: v
      });
    },
    setMapContainer: (v) => {
      P({
        type: "SET_MAP_CONTAINER",
        payload: v
      });
    }
  };
  return /* @__PURE__ */ K(Ye.Provider, { value: b, children: s.children });
}, kr = Ne(
  ({
    options: s,
    mapKey: f,
    serviceKey: P = "",
    zoom: b = 12,
    defaultType: v = "neshan",
    scale: F,
    center: A,
    poi: O = !0,
    traffic: m = !0,
    onInit: x,
    className: _,
    style: T
  }, j) => {
    const C = mr(Ye), S = yr(null), [Y, H] = ce(!1), [w, y] = ce(null), [fe, X] = ce();
    hr(
      j,
      () => ({
        get mapContainer() {
          return S;
        },
        get ol() {
          return typeof ol < "u" ? ol : void 0;
        },
        get map() {
          return C.map;
        },
        get isInitiated() {
          return Y;
        }
      }),
      [Y, C.map]
    );
    const $ = () => {
      var g, l;
      if (w)
        return w;
      const u = (l = (g = S.current) == null ? void 0 : g.querySelector(".ol-viewport")) == null ? void 0 : l.querySelectorAll("div:last-child");
      if (u)
        return y(
          Array.from(u)[u.length - 1]
        ), w;
    }, Z = () => {
      const u = $();
      u == null || u.style.setProperty("scale", "" + F);
    }, Q = () => {
      const u = $();
      u == null || u.style.setProperty("bottom", "0.2em"), u == null || u.style.setProperty("left", "0.4em");
    }, ee = (u, g = "my-openlayer") => {
      const l = document.getElementById(g);
      if (l)
        return l;
      const h = document.createElement("script");
      return h.src = u, h.id = g, document.getElementsByTagName("head")[0].appendChild(h), h;
    }, V = async () => {
      var h;
      if (!S.current)
        return;
      const u = We(A);
      X(u);
      const g = u || await wr(), l = new ol.Map({
        target: S.current,
        key: f || "",
        mapType: v,
        // poi: store.state.poiLayer,
        poi: O,
        // traffic: store.state.trafficLayer,
        traffic: m,
        view: new ol.View({
          center: ol.proj.fromLonLat(g),
          zoom: b
          // Not in ol@5
          // smoothExtentConstraint: true,
          // projection: 'EPSG:4326' //Default was EPSG:3857
        }),
        controls: [],
        ...s
      });
      l.setMapType(v), C.setMap(l), (h = C.map) == null || h.setMapType(v), C.setMapType(v), H(!0), x && x(ol, l);
    }, B = () => {
      let u, g, l;
      return u = new Promise((h, D) => {
        l = D, g = h, V().then(() => {
          Z(), Q(), g();
        }).catch(() => {
          l();
        });
      }), { promise: u, cancelPromise: () => l() };
    };
    return gr(() => {
      if (S.current && C.setMapContainer(S.current), typeof ol < "u") {
        const u = B();
        return () => u.cancelPromise();
      } else {
        const u = ee(Or.map), g = () => {
          B();
        };
        return u.addEventListener("load", g), () => u.removeEventListener("load", g);
      }
    }, []), /* @__PURE__ */ K(
      "div",
      {
        ref: S,
        className: _ || Sr.map,
        style: T
      }
    );
  }
), Dr = Ne(
  (s, f) => /* @__PURE__ */ K(jr, { children: /* @__PURE__ */ K(kr, { ref: f, ...s }) })
);
export {
  Dr as default
};
